src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -1903,6 +2009,39 @@ void QgsAdvancedDigitizingDockWidget::addPoint( const QgsPointXY &point )
     mCadPointList.insert( 0, pt );
   }
 
+  if ( recordConstructionGuides() )
+  {
+    if ( constructionMode() )
+    {
+      mConstructionGuideLine.addVertex( pt );
+
+      if ( mConstructionGuideLine.numPoints() == 2 )
+      {
+        QgsFeature feature;
+        QgsGeometry geom( mConstructionGuideLine.clone() );
+        feature.setGeometry( geom );
+        mConstructionGuidesLayer->addFeature( feature, QgsFeatureSink::FastInsert );
```

Comment:
We don't require the extra checks and undo stack logic for the vector layer edits here, so it'd be more efficient to just

```suggestion
        mConstructionGuidesLayer->dataProvider()->addFeature( feature, QgsFeatureSink::FastInsert );
```

and avoid making the layer editable entirely (removes the need for a useless edit buffer)